components/sup/src/config.rs

```diff
@@ -30,6 +32,32 @@ use topology::Topology;
 
 static LOGKEY: &'static str = "CFG";
 
+/// The Static Global Configuration.
+///
+/// This sets up a raw pointer, which we are going to transmute to a Box<Config>
+/// with the first call to gcache().
+static mut CONFIG: *const Config = 0 as *const Config;
+
+/// Store a configuration, for later use through `gconfig()`.
+///
+/// MUST BE CALLED BEFORE ANY CALLS TO `gconfig()`.
+pub fn gcache(config: Config) {
```

Comment:
Right now, the Config struct is immutable after it is created. It's passed around as a reference, without mutability. 

If we decide we want it to be mutable, we will need to refactor into having the static hold a `RwLock<Config>` instead. Given that right now, for the supervisor at least, these are only options populated from the CLI, we're fine to let it remain immutable.